setup_python.sh

```diff
@@ -38,20 +32,32 @@ case $DD_TARGET_ARCH in
         libncursesw5-dev libssl-dev libsqlite3-dev tk-dev libgdbm-dev \
         libc6-dev libbz2-dev libffi-dev zlib1g-dev
     elif [ -f /etc/redhat-release ] || [ "$DISTRIBUTION" == "RedHat" ] || [ "$DISTRIBUTION" == "CentOS" ] || [ "$DISTRIBUTION" == "Amazon" ]; then
-        yum install -y gcc openssl-devel bzip2-devel libffi-devel wget make
+        yum install -y gcc openssl-devel bzip2-devel libffi-devel wget make perl-IPC-Cmd
     fi
-
-    wget https://www.python.org/ftp/python/3.9.5/Python-3.9.5.tgz
-    echo "e0fbd5b6e1ee242524430dee3c91baf4cbbaba4a72dd1674b90fda87b713c7ab Python-3.9.5.tgz" | sha256sum --check
-    tar xzf Python-3.9.5.tgz
-    pushd /Python-3.9.5
-        ./configure
-        make -j 8
+    OPENSSL_VERSION="3.0.12"
+    OPENSSL_SHA256="f93c9e8edde5e9166119de31755fc87b4aa34863662f67ddfcba14d0b6b69b61"
+    PYTHON_SHA256="a12a0a013a30b846c786c010f2c19dd36b7298d888f7c4bd1581d90ce18b5e58"
+
+    wget https://ftp.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz
+    echo "$OPENSSL_SHA256 openssl-$OPENSSL_VERSION.tar.gz" | sha256sum --check
+    tar xzf openssl-$OPENSSL_VERSION.tar.gz
+    pushd openssl-$OPENSSL_VERSION
+      ./config --prefix=/opt/openssl --openssldir=/usr/local/ssl -Wl,-Bsymbolic-functions -fPIC shared no-ssl2 no-ssl3 linux-generic32
+      make -j $(nproc)
+      make install
+    popd
+    rm -rf openssl-$OPENSSL_VERSION.tar.gz openssl-$OPENSSL_VERSION
+
+    wget https://www.python.org/ftp/python/$PY3_VERSION/Python-$PY3_VERSION.tgz
+    echo "$PYTHON_SHA256 Python-$PY3_VERSION.tgz" | sha256sum --check
+    tar xzf Python-$PY3_VERSION.tgz
+    pushd /Python-$PY3_VERSION
+        ./configure --with-openssl=/opt/openssl --with-openssl-rpath=auto
+        make -j $(nproc)
         make install
     popd
-    rm -rf Python-3.9.5
-    rm Python-3.9.5.tgz
-    ln -sf /usr/bin/python3.9 /usr/bin/python3
+    rm -rf Python-$PY3_VERSION Python-$PY3_VERSION.tgz
+    ln -sf /usr/bin/python3.11 /usr/bin/python3
```

Comment:
Is this meant to be the python executable you just built? Or another version installed some other way?

Reply:
Yes it is the Python executable we just built

Reply:
Then I think something is wrong, by default (ie. without `--prefix`) python should be installed in `/usr/local/`

Reply:
I'll pull the image and double check then! But if that's wrong, it's already broken 😅

Reply:
you're right, we are using /usr/local/bin/python3 and `/usr/bin/python3` is a broken link for the exact reason you mention. I'll drop the `ln`

Reply:
Thanks for checking! I was a bit confused as to why it worked if the executable wasn't there, but if it uses it from /usr/local directly then it makes sense :)